src/gitHubHelper.ts

```diff
@@ -0,0 +1,98 @@
+import * as core from '@actions/core';
+import * as dockleHelper from './dockleHelper';
+import * as fileHelper from './fileHelper';
+import * as inputHelper from './inputHelper';
+import * as trivyHelper from './trivyHelper';
+import { WebRequest, WebResponse, sendRequest } from "./httpClient";
+
+export async function createCheckRunWithScanResult(trivyStatus: number, dockleStatus: number): Promise<void> {
+    const checkRunUrl = `https://api.github.com/repos/${process.env.GITHUB_REPOSITORY}/check-runs`;
+    const gitHubToken = core.getInput("github-token");
+    const pullRequestHeadSha = getPullRequestHeadSha();
+    const checkConclusion = getCheckConclusion(trivyStatus, dockleStatus);
+    const checkSummary = getCheckSummary(trivyStatus, dockleStatus);
+    const checkText = getCheckText(trivyStatus, dockleStatus);
+
+    const checkRunPayload = {
+        head_sha: pullRequestHeadSha,
+        name: inputHelper.imageName,
+        status: "completed",
+        conclusion: checkConclusion,
+        output: {
+            title: "Container scan result",
+            summary: checkSummary,
+            text: checkText
+        }
+    }
+
+    core.debug(`Creating check run with scan result against head_sha: ${pullRequestHeadSha}...`);
+
+    let webRequest = new WebRequest();
+    webRequest.method = "POST";
+    webRequest.uri = checkRunUrl;
+    webRequest.body = JSON.stringify(checkRunPayload);
+    webRequest.headers = {        
+            Authorization: `Bearer ${gitHubToken}`,
+            Accept: 'application/vnd.github.antiope-preview+json'
+    };
+
+    const response: WebResponse = await sendRequest(webRequest);
+    if (response.statusCode != 201) {
+        throw Error(`Statuscode: ${response.statusCode}, StatusMessage: ${response.statusMessage}, Url: ${checkRunUrl}, head_sha: ${pullRequestHeadSha}`);
+    }
+
+    console.log(`Created check run with container scan results.\nId: ${response.body['id']}, Url: ${response.body['html_url']}`)
+}
+
+function getCheckConclusion(trivyStatus: number, dockleStatus: number): string {
+    let checkConclusion: string;
+    if(trivyStatus != 0 || (dockleHelper.isCisChecksEnabled() && dockleStatus != 0)) {
+        checkConclusion = 'failure';
+    } else {
+        checkConclusion = 'success';
+    } 
+
+    return checkConclusion;
+}
+
+function getPullRequestHeadSha(): string {
+    const eventJson = getEventJson();
+    return eventJson["pull_request"]["head"]["sha"];
```

Comment:
Add null checks. It probably wouldn't exist in non-PR cases

Reply:
Pull_request's head sha would be same as `GITHUB_SHA`, right?

Reply:
Changed logic to return `eventJson["pull_request"]["head"]["sha"]` when triggering event is pull_request`, and `GITHUB_SHA` otherwise.
Note: GITHUB_SHA is merge commit in case of pull request.